src/main/java/com/todolist/app/model/Item.java

```diff
@@ -12,12 +11,12 @@ public class Item {
     private Priority priority;
     private String date;
     private List<Tag> tags;
-    public Item(String value, Status status, Priority priority, String date){
+    public Item(String value, String date, Status status, Priority priority, ArrayList<Tag> tags){
```

Comment:
One of the best practices is to use interfaces instead of direct implementation.

```
Using interfaces over concrete types is the key for good encapsulation and for loose coupling your code.
```

Link: https://stackoverflow.com/questions/147468/why-should-the-interface-for-a-java-class-be-preferred